src/Compilers/Core/Portable/TreeDumper.cs

```diff
@@ -106,6 +102,31 @@ private void DoDumpCompact(TreeDumperNode node, string indent)
                 // the child node's children:
                 DoDumpCompact(child, indent + (i == children.Count - 1 ? "  " : "\u2502 "));
             }
+
+            static IEnumerable<TreeDumperNode> filter(IEnumerable<TreeDumperNode> nodes)
```

Comment:
:memo: this is not essential to the PR, but greatly helps when staring with large bound trees